openapi.yaml

```diff
@@ -0,0 +1,390 @@
+  termsOfService: https://www.selfdefined.app/terms-of-service
+  # contact:
+  #   email: apiteam@swagger.io
+  # license:
+  #   name: Apache 2.0
+  #   url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+externalDocs:
+  description: Find out more
+  url: https://github.com/tatianamac/selfdefined
+servers:
+- url: https://api.selfdefined.app/
+tags:
+- name: Word
+  description: Self-Defined Words
+  externalDocs:
+    description: Find out more
+    url: https://schemas.selfdefined.app/words
+components:
+  schemas:
+    Word:
+      allOf: 
+      - $ref: '#/components/schemas/Id'
+      - $ref: '#/components/schemas/NewWord'
+    FullWord:
+      allOf: 
+      - $ref: '#/components/schemas/Word'
+      - $ref: '#/components/schemas/Translations'
+      - $ref: '#/components/schemas/Synonyms'
+      - $ref: '#/components/schemas/Alternatives'
+      - $ref: '#/components/schemas/FurtherReadings'
+    ShortLink:
+      description: Shortlink terms which map to this word.
+      type: object
+      properties:
+        shortLinkText:
+          type: string
+        word:
+          $ref: '#/components/schemas/Link'
+      required:
+      - shortLink
+      - word
+    Link:
+      type: object
+      properties:
+        displayName:
+          type: string
+        url:
+          type: string
+          format: url
+        icon:
+          type: string
+          format: url
+      required:
+      - displayName
+      - url
+    Translations:
+      type: object
+      properties:
+        translations:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/Link'
+    Synonyms:
+      type: object
+      properties:
+        synonyms:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/Link'
+    Alternatives:
+      description: A positive alternative choice to communicate without exclusion.
+      type: object
+      properties:
+        alternatives:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/Link'
+    FurtherReadings:
+      description: For further reading on the word.
+      type: object
+      properties:
+        furtherReadings:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/Link'
+    NewWord:
+      type: object
+      properties:
+        displayName:
+          type: string
+        definition:
+          type: string
+        originLanguage:
+          type: string
+          description: This should be the ISO internationalization value.
+        tag:
```

Comment:
Tag should be an array because multiple tags might apply. I like the enum but it might be better to allow it to be dynamic and let the API implementation and/or web front end enforce rules around how they're created/added to the word.